### PR TITLE
chore: group all dependabot upgrades

### PR DESCRIPTION
**Pull request by petertonysmith94**
# Summary

- Further groups dependabot upgrades by patch/minor and major. 
- This _should_ alleviate the issue we're experiencing when once we merge a single dependabot PR, we _usually_ have conflicts in all subsequential PR's

# Checklist

- [X] All **changes** are **covered** by **tests** (or not applicable)
- [X] All **changes** are **documented** (or not applicable)
- [X] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [X] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

**Review by Torres-ssf — APPROVED** [review]

File: .github/dependabot.yml [hunk 1]
```diff
@@ -10,20 +10,12 @@ updates:
       - "chore"
     commit-message:
       prefix: "build(deps):"
-      prefix-development: "build(deps-dev):"
     versioning-strategy: "increase"
     groups:
-      dev-deps:
-        dependency-type: "development"
+      all:
```
Comment (arboleya) [inline .github/dependabot.yml:15]: Should we at least group `patch` updates from the rest? The rationale is that patches should always be backward-compatible and one click away, whereas minors/majors may need manual intervention, which could block the whole PR.

Comment (danielbate) [reply .github/dependabot.yml]: https://github.com/FuelLabs/fuels-ts/pull/2976/commits/35debaa057b7903279d6620f9ae78b8e4b7b7bb1

**Review by danielbate — APPROVED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (arboleya) [reply .github/dependabot.yml]: This way, we'll end up with four groups, right?

Could we achieve having only two?

1. deps + deps dev (patches)
1. deps + deps dev (major + minor)

Comment (danielbate) [reply .github/dependabot.yml]: Correct, the docs sounds like it's going to split them so we lose the prefixing, but that's not necessarily a problem. And I might be wrong. [Let's see](https://github.com/FuelLabs/fuels-ts/pull/2976/commits/fbc2f587f74a499b2358066c91a5ae18d1d7eac9).

**Review by danielbate — APPROVED** [review]

**Review by arboleya — APPROVED** [review]

**Review by Torres-ssf — APPROVED** [review]

**Pull request merged**